packages/sources/web-components/src/components/vtmn-badge/vtmn-badge.tsx

```diff
@@ -0,0 +1,40 @@
+import { Component, Prop, h, ComponentInterface } from '@stencil/core';
+
+@Component({
+  tag: 'vtmn-badge',
+  styleUrl: 'vtmn-badge.pcss',
+})
+export class VtmnBadge implements ComponentInterface {
+  /**
+   * The variant of the badge.
+   * @default null
+   */
+  @Prop() variant: 'default' | 'brand' | 'reversed' | 'accent';
+
+  /**
+   * The size of the badge.
+   * @default null
+   */
+  @Prop() size: 'small' | 'medium';
+
+  /**
+   * The target of the badge
+   * @default null
+   */
+  @Prop() value: string;
+
+  render() {
+    return (
+      <div
+        class={[
+          'vtmn-badge',
```

Comment:
suggestion: use a dedicated lib for that kind of class constructions, it will help you avoid silly errors
ex classnames  or clsx

```tsx
<div class={classnames('vtmn-badge', {
  [`vtmn-badge_size--${this.size}`]: this.size,
  [`vtmn-badge_variant--${this.variant}`]: this.variant,
})}></div>
```

Reply:
Thanks for your feedback, unfortunately I didn't find examples of class constructions for web components. Do you have a suggestion to make ? (Your example given doesn't work with stencil)